internal/delete_util.go

```diff
@@ -10,27 +10,14 @@ import (
 	"github.com/wal-g/wal-g/utility"
 )
 
-func FindPermanentBackups(folder storage.Folder, metaFetcher GenericMetaFetcher) map[string]bool {
-	tracelog.InfoLogger.Println("retrieving permanent objects")
-	backupTimes, err := GetBackups(folder.GetSubFolder(utility.BaseBackupPath))
-	if err != nil {
-		return map[string]bool{}
-	}
-
-	permanentBackups := map[string]bool{}
-	for _, backupTime := range backupTimes {
-		meta, err := metaFetcher.Fetch(
-			backupTime.BackupName, folder.GetSubFolder(utility.BaseBackupPath))
-		if err != nil {
-			tracelog.ErrorLogger.Printf("failed to fetch backup meta for backup %s with error %s, ignoring...",
-				backupTime.BackupName, err.Error())
-			continue
-		}
-		if meta.IsPermanent {
-			permanentBackups[backupTime.BackupName] = true
-		}
+func PrintMetadataNotFoundError(backupTime BackupTime, err error) {
```

Comment:
well, now this function not only prints but exits in case of non-404s so I guess it needs to be renamed